{{cookiecutter.package_name}}/src/{{cookiecutter.package_name}}/tools/cli.py

```diff
@@ -6,136 +6,159 @@
+        Proper location names should come from GBD location set 1, the
+        location reporting hierarchy wherever possible. If using sub-national
+        locations, they may be found in GBD location set 35, the model
+        results location hierarchy.
+
+    Yields
+    -------
+        Named tuples with both the proper and sanitized location names
+
+    Examples
+    --------
+    >>> sanitize("Nigeria")
+    Location(proper="Nigeria", sanitized="nigeria")
 
+    >>> sanitize("Burkina Faso")
+    Location(proper="Burkina Faso", sanitized="burkina_faso")
 
-def sanitize(loc: str) -> (bool, Location):
-    loc_proper = loc.strip()
-    if loc_proper.startswith('#'):
-        loc_proper = loc_sanitized = ''
-    else:
-        loc_sanitized = loc_proper
-        loc_sanitized = re.sub("[- ,.&']", '_', loc_sanitized).lower()
-    return bool(loc_proper), Location(loc_proper, loc_sanitized)
+    >>> sanitize("Cote d'Ivoire")
+    Location(proper="Cote d'Ivoire", sanitized="cote_d-ivoire")
 
+    """
+    # TODO: Check that they're in a call to get_location_id from
+    #    vivarium_gbd_access.gbd.  Not doing this now because it's not
+    #    specified as a proper dependency in the setup.py and would be a
+    #    bit of a pain to do now.
+    for location in locations:
+        proper = location.strip()
+        sanitized = re.sub("[- ,.&']", '_', proper).lower()
+        yield Location(proper, sanitized)
+
+
+def parse_locations(locations_file: Optional[str], single_location: Optional[str]) -> List[str]:
+    """Parses location inputs into a list of location strings.
+
+    If no arguments are provided, this will default to the ``locations.txt``
+    file located in the repository model_specifications directory.
+
+    Parameters
+    ----------
+    locations_file
+        Path to a file containing a list of locations to generate model
+        specifications for.
+    single_location
+        Optional single location to generate a model specification for.
+
+    Returns
+    -------
+        A list of location strings.
+
+    Raises
+    ------
+    ValueError
+        If both ``locations_file`` and ``single_loc`` are provided or if
+        a ``locations_file`` with no locations is provided.
 
-def get_sanitized_locations(single_loc: str, loc_file: Path) -> List[Location]:
-    """ Naming convention is to lowercase the location string, replace spaces with underscores,
-        and capitalize the first letter. No manipulation happens in the template code.
     """
-    def helper(item):
-        valid, loc = sanitize(item)
-        if valid:
-            locs.append(loc)
-
-    locs = []
-    if single_loc:
-        helper(single_loc)
-    else:
-        with loc_file.open('r') as infile:
-            for line in infile:
-                helper(line)
-    return locs
-
-
-def log_and_exit(f: Callable) -> Callable:
-    @wraps(f)
-    def wrapper(*args, **kwds):
-        f(*args, **kwds)
-        exit(1)
```

Comment:
The only good time to log errors is if there is an expectation that the error can be caught and handled in a non-fatal manner by a client tool.  Otherwise, always raise an exception. This will produce the proper exit code by itself:

```
collijk@smartypants:~$ python -c "raise Exception"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
Exception
collijk@smartypants:~$ echo $?
1
```

Reply:
loguru, in particular, is excellent about providing really robust stack traces from raised exceptions, which makes our debugging job a lot easier (if we can convince the research to post their stack traces)